handler.js

```diff
@@ -360,117 +370,230 @@ class GithubBuild {
     // parse [force uat|functional <argument1> ... <argumentN>]
-    const commandRegex = new RegExp(/^.*\[force (uat|functional)(\s[^\]]+)?\]/m);
+    const commandRegex = new RegExp(/^.*\[force (uat|functional|deployment)(\s[^\]]+)?\]/m);
     let matches = commandRegex.exec(this.getCommitMsg());
 
     if (matches === null || matches.length < 3) {
       return null;
     }
 
     return [
-      // matched "uat" or "functional"
+      // matched "uat", "functional" or "deployment"
       matches[1],
 
       // matched argument
       matches[2]
     ];
   }
 
-  getTests() {
-    let tests = [];
-    const branch = this.getBranch();
+  checkForReleaseOrMasterBranches() {
+    return {
+      isMatch: () => {
+        const {branch} = this.dataSet;
+        return ["master", "release"].includes(branch);
+      },
+      getTests: () => {
+        return [{
+          name: "js-php",
+          type: "unit-tests",
+          deployable: false
+        }];
+      }
+    }
+  };
 
-    if(branch === "release" || branch === "master") {
-      tests.push({
-        name: "js-php",
-        type: "unit-tests",
-        deployable: false
-      });
-      // no need to run tests when pushing to release
-      // should be only the case when merging a PR on release
-    } else if (this.enableUatAndFunctionalTests()) {
-      // Only enabled for PR against release
-
-      console.log("deploy and run all tests on release");
-      tests.push({
-        name: "js-php",
-        type: "unit-tests",
-        deployable: false
-      });
-      tests.push({
-        name: "backend",
-        type: "uat",
-        deployable: true
-      });
+  checkForOtherBranchesOfUATFunctionalTestsEnabled() {
+    return {
+      isMatch: () => {
+        const {branch, isEnabledUatFunctional} = this.dataSet;
+        return !["master", "release"].includes(branch) && isEnabledUatFunctional;
+      },
+      getTests: () => {
+        return [
+          {
+            name: "js-php",
+            type: "unit-tests",
+            deployable: false
+          },
+          {
+            name: "backend",
+            type: "uat",
+            deployable: false
+          },
+          {
+            name: "frontend",
+            type: "uat",
+            deployable: false
+          },
+          {
+            name: "functional",
+            type: "functional",
+            deployable: false
+          }
+        ];
+      }
+    }
+  };
 
-      tests.push({
-        name: "frontend",
-        type: "uat",
-        deployable: false
-      });
+  checkForOtherBranchesOfUATFunctionalTestsNotEnabled() {
+    return {
+      isMatch: () => {
+        const {branch, isEnabledUatFunctional} = this.dataSet;
+        return !["master", "release"].includes(branch) && !isEnabledUatFunctional;
+      },
+      getTests: () => {
+        let response = [];
+        // run unit-tests only if there is no skip command
+        if (this.checkForSkipUnitTestsCommand().isMatch()) {
```

Comment:
The rules in here should be flattened out and included the main getTests function for simplicity, the isMatch of this class can be reused - in that case, this function should be a class, other check functions should be changed to class as well, may be introduced `Rule` class/interface with 2 default methods `isMatch` and `getTests`?